pkg/latte/intent_reauthenticate.go

```diff
@@ -0,0 +1,88 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/session"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentReauthenticate{})
+}
+
+var IntentReauthenticateSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false
+	}
+`)
+
+type IntentReauthenticate struct {
+}
+
+func (*IntentReauthenticate) Kind() string {
+	return "latte.IntentReauthenticate"
+}
+
+func (*IntentReauthenticate) JSONSchema() *validation.SimpleSchema {
+	return IntentReauthenticateSchema
+}
+
+func (*IntentReauthenticate) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	switch len(w.Nodes) {
+	case 0:
+		return nil, nil
+	case 1:
+		return nil, nil
+	}
+
+	return nil, workflow.ErrEOF
+}
+
+func (i *IntentReauthenticate) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	userID, err := i.userID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	switch len(w.Nodes) {
+	case 0:
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This err checking block is redundant.

Reply:
Handled in [ae3db4b](https://github.com/authgear/authgear-server/pull/3322/commits/ae3db4b3079c7174b8f18f6c7dd42ad2603d9f1d)